src/com/android/calendar/month/MonthWeekEventsView.java

```diff
@@ -785,152 +753,912 @@ protected int addChipOutline(FloatRef lines, int count, int x, int y) {
+                mMinItems = Math.max(mMinItems, yIndex);
+            }
+            mListSize = Math.max(mMinItems + 1, eventsHeight);
+            mRemainingEvents.clear();
+        }
+
+        /**
+         * Returns index of next slot in FormattedEventBase Array.
+         * @param indexedEvents
+         * @param index
+         * @return index of next slot
+         */
+        protected int getNextIndex(FormattedEventBase[] indexedEvents, int index) {
+            if (index < mMinItems) {
+                return index + 1;
+            }
+            return index + indexedEvents[index].getFormat().getEventLines();
+        }
+
+        protected FormattedEventBase[] fillInIndexedEvents(ArrayList<FormattedEventBase> dayEvents) {
+            FormattedEventBase[] indexedEvents = new FormattedEventBase[mListSize];
+            for (FormattedEventBase event : dayEvents) {
+                if (event.getFormat().getYIndex() != -1) {
+                    indexedEvents[event.getFormat().getYIndex()] = event;
+                } else {
+                    sortedAddRemainingEventToList(mRemainingEvents, event);
+                }
+            }
+            return indexedEvents;
+        }
+
+        protected ArrayList<FormattedEventBase> getSortedEvents(FormattedEventBase[] indexedEvents,
+                                                            int expectedSize) {
+            ArrayList<FormattedEventBase> sortedEvents = new ArrayList<>(expectedSize);
+            for (FormattedEventBase event : indexedEvents) {
+                if (event != null) {
+                    sortedEvents.add(event);
+                }
+            }
+            return sortedEvents;
+        }
+
+        protected void fillInRemainingEvents(FormattedEventBase[] indexedEvents) {
+            int index = 0;
+            for (FormattedEventBase event : mRemainingEvents) {
+                if (!event.getFormat().isVisible()) {
+                    continue;
+                }
+                while (index < indexedEvents.length) {
+                    if (indexedEvents[index] == null) {
+                        event.getFormat().setYIndex(index);
+                        if (index < mMinItems) {
+                            event.getFormat().capEventLinesAt(1);
+                            if (!event.isBordered()) {
+                                event.setBoundaries(mFixedHeightBoundaries);
+                            }
+                        }
+                        indexedEvents[index] = event;
+                        index = getNextIndex(indexedEvents, index);
+                        break;
+                    }
+                    index = getNextIndex(indexedEvents, index);
+                }
+            }
+            addVirtualEvents(indexedEvents, index);
+        }
+
+        protected void addVirtualEvents(FormattedEventBase[] indexedEvents, int initialIndex)  {
+            for (int index = initialIndex; index < mMinItems; ++index) {
```

Comment:
I would go for `index++` (seems to be more common in the code base).

Reply:
using pre incrementation is more efficient (but compiler should figure it out and change post-incrementation to pre-incrementation whenever it has no side effect)
I have changed the code to follow your suggestion